code/game/objects/items/weapons/tools/upgrades.dm

```diff
@@ -0,0 +1,445 @@
+	required_qualities = list(QUALITY_CUTTING,QUALITY_SAWING, QUALITY_WIRE_CUTTING, QUALITY_PRYING)
+	prefix = "diamond-edged"
+	workspeed = 0.25
+	degradation_mult = 0.85
+	force_mult = 1.10
+	matter = list(MATERIAL_STEEL = 1, MATERIAL_DIAMOND = 1)
+
+
+/obj/item/weapon/tool_upgrade/productivity/oxyjet
+	name = "oxyjet canister"
+	desc = "A canister of pure, compressed oxygen with adapters for mounting onto a welding tool. Used alongside fuel, it allows for higher burn temperatures"
+	icon_state = "oxyjet"
+	required_qualities = list(QUALITY_WELDING)
+	prefix = "oxyjet"
+	workspeed = 0.20
+	force_mult = 1.15
+
+
+//Enhances power tools majorly, but also increases costs
+/obj/item/weapon/tool_upgrade/productivity/motor
+	name = "high power motor"
+	desc = "A motor for power tools with a higher horsepower than usually expected. Significantly enhances productivity and lifespan, but more expensive to run and harder to control"
+	icon_state = "motor"
+	required_qualities = list(QUALITY_SCREW_DRIVING, QUALITY_DRILLING, QUALITY_SAWING, QUALITY_DIGGING, QUALITY_EXCAVATION)
+	prefix = "high-power"
+	workspeed = 0.50
+	degradation_mult = 0.75
+	force_mult = 1.15
+
+	powercost_mult = 1.5
+	fuelcost_mult = 1.5
+	precision = -10
+
+/obj/item/weapon/tool_upgrade/productivity/motor/can_apply(var/obj/item/weapon/tool/T, var/mob/user)
+	.=..()
+	if (.)
+		if (T.use_fuel_cost || T.use_power_cost)
+			return TRUE
+		return FALSE
+
+
+
+
+
+
+// 	 REFINEMENT: INCREASES PRECISION
+//------------------------------------------------
+/obj/item/weapon/tool_upgrade/refinement/laserguide
+	name = "Asters \"Guiding Light\" laser guide"
+	desc = "A small visible laser which can be strapped onto any tool, giving an accurate representation of its target. Helps improve precision"
+	icon_state = "laser_guide"
+	prefix = "laser-guided"
+	precision = 10
+	matter = list(MATERIAL_PLASTIC = 1, MATERIAL_URANIUM = 1)
+
+
+//Fits onto generally small tools that require precision, especially surgical tools
+//Doesn't work onlarger things like crowbars and drills
+/obj/item/weapon/tool_upgrade/refinement/stabilized_grip
+	name = "gyrostabilized grip"
+	desc = "A fancy mechanical grip that partially floats around a tool, absorbing tremors and shocks. Allows precise work with a shaky hand"
+	icon_state = "stabilizing"
+	required_qualities = list(QUALITY_CUTTING,QUALITY_WIRE_CUTTING, QUALITY_SCREW_DRIVING, QUALITY_WELDING,
+	QUALITY_PULSING, QUALITY_CLAMPING, QUALITY_CAUTERIZING, QUALITY_BONE_SETTING, QUALITY_LASER_CUTTING)
+	prefix = "stabilized"
+	precision = 10
+
+/obj/item/weapon/tool_upgrade/refinement/magbit
+	name = "magnetic bit"
+	desc = "Magnetises tools used for handling small objects, reducing instances of dropping screws and bolts."
+	icon_state = "magnetic"
+	required_qualities = list(QUALITY_SCREW_DRIVING, QUALITY_BOLT_TURNING, QUALITY_CLAMPING, QUALITY_BONE_SETTING)
+	prefix = "magnetic"
+	precision = 10
+
+
+/obj/item/weapon/tool_upgrade/refinement/ported_barrel
```

Comment:
This upgrade needs more degradation multiplier, to make tool with this installed less reliable.